plugins/urls/handlers/handler.py

```diff
@@ -0,0 +1,149 @@
+__author__ = 'Gareth Coles'
+
+import re
+
+
+class URLHandler(object):
+    """
+    URL handler. Subclass this!
+
+    You'll want to override both the `call` method and the *criteria* dict.
+    The former is called if the criteria matches the URL.
+
+    In the criteria dict, you're expected to provide values to test equality
+    for. However, there are a few things to be aware of.
+
+    * Leave a key out if you don't care about matching it - None will be
+      matched against.
+    * You may provide a compiled regular expression to test against as well.
+    * Finally, you may provide a callable (function or class), which will be
+      run for the comparison instead, and should return either True or False.
+
+    >>> criteria = {
+    ...     # No protocol here, since we don't care about it
+    ...     "auth": lambda x: x is not None
+    ...     "domain": re.compile(u"[a-zA-Z]+"),
+    ...     "port": lambda x: x > 8080,
+    ...     "path": lambda x: len(x) > 10
+    ... }
+    ...
+    >>>
+
+    Additionally, if the above matching is somehow not good enough for you, you
+    may override the `match` function.
+    """
+
+    # Apparently, different implementations use different base classes,
+    # so this is how we should do this.
+    # I've been told hasattr(obj, "match") would work too, but, well, this
+    # feels like too much of a kludge, even for me. Plus, there are other
+    # objects that have a "match" function.
+    __REGEX_TYPE = type(re.compile(""))
+
+    # Remember to set this, so that there are no conflicting handlers - only
+    # one handler per name!
+    name = ""
+
+    plugin = None
+    urls_plugin = None
+
+    criteria = {
```

Comment:
Mutable instance variables set in class definition.

Also, the docstring above says None will be matched against, so I'm not sure what the point of this is. To make nothing match this handler unless you replace it in the subclass?